### PR TITLE
OSDOCS-6617: update version-specific info MicroShift

### PR DESCRIPTION
**Pull request by ShaunaDiaz**
Version(s):
4.14+

Issue:
https://issues.redhat.com/browse/OSDOCS-6617

#### remove TP snippet, update version as needed
https://63358--docspreview.netlify.app/microshift/latest/microshift_install/microshift-install-rpm
https://63358--docspreview.netlify.app/microshift/latest/microshift_install/microshift-embed-in-rpm-ostree
https://63358--docspreview.netlify.app/microshift/latest/microshift_support/microshift-etcd
https://63358--docspreview.netlify.app/microshift/latest/microshift_support/microshift-sos-report
#### version updates in general
https://63358--docspreview.netlify.app/microshift/latest/microshift_rest_api/understanding-compatibility-guidelines#api-compatibility-common-terminology-releases_compatibility-guidelines
https://63358--docspreview.netlify.app/microshift/latest/microshift_install/microshift-embed-in-rpm-ostree#adding-microshift-service-to-blueprint_microshift-embed-in-rpm-ostree
https://63358--docspreview.netlify.app/microshift/latest/microshift_troubleshooting/microshift-version#microshift-version-api_microshift-version
#### new attributes
https://63358--docspreview.netlify.app/microshift/latest/microshift_install/microshift-embed-in-rpm-ostree#adding-microshift-repos-image-builder_microshift-embed-in-rpm-ostree
https://63358--docspreview.netlify.app/microshift/latest/microshift_install/microshift-embed-in-rpm-ostree#microshift-creating-ostree-iso_microshift-embed-in-rpm-ostree

QE review:
- [x] QE has approved this change.

### CODE REVIEW

File: modules/api-compatibility-guidelines.adoc [hunk 1]
```diff
@@ -9,7 +9,7 @@ Red Hat recommends that application developers adopt the following principles in
 
 * Use APIs and components with support tiers that match the application's need.
 * Build applications using the published client libraries where possible.
-* Applications are only guaranteed to run correctly if they execute in an environment that is as new as the environment it was built to execute against. An application that was built for {product-title} 4.7 is not guaranteed to function properly on {product-title} 4.6.
+* Applications are only guaranteed to run correctly if they execute in an environment that is as new as the environment it was built to execute against. An application that was built for {product-title} 4.13 is not guaranteed to function properly on {product-title} 4.14.
```
[the remark this answers is not in the document]
Comment (ShaunaDiaz) [reply modules/api-compatibility-guidelines.adoc:12]: Swapped the numbers. There is no MicroShift 4.6 or 4.6, so it doesn't make sense; not are the OCP versions referenced still supported. Thanks for the catch!

**Comment by ShaunaDiaz** [comment]
@dhensel-rh Could you PTAL? Thank you!

**Comment by ShaunaDiaz** [comment]
@dhellmann lgtm now?

**Review by dhellmann — COMMENTED** [review]
/lgtm

File: microshift_storage/dynamic-provisioning-microshift.adoc [hunk 2]
```diff
@@ -6,6 +6,8 @@ include::_attributes/attributes-microshift.adoc[]
 
 toc::[]
 
+Learn about dynamic provisioning for {product-title} including setting storage classes and associated configuration.
```
Comment (kcarmichael08) [inline microshift_storage/dynamic-provisioning-microshift.adoc:9]:
```suggestion
Learn about dynamic provisioning for {product-title}, including setting storage classes and associated configuration.
```

File: microshift_storage/generic-ephemeral-volumes-microshift.adoc [hunk 3]
```diff
@@ -6,6 +6,8 @@ include::_attributes/common-attributes.adoc[]
 
 toc::[]
 
+Learn about ephemeral volumes for {product-title} including their lifecycles, security, and naming.
```
Comment (kcarmichael08) [inline microshift_storage/generic-ephemeral-volumes-microshift.adoc:9]:
```suggestion
Learn about ephemeral volumes for {product-title}, including their lifecycles, security, and naming.
```

**Review by kcarmichael08 — COMMENTED** [review]
Very minor suggestion; otherwise lgtm.

**Pull request merged**